example/src/Keyboard/KeyboardPage.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+import { Alert, StyleSheet, TextInput as TextInputRN, View } from 'react-native';
+import { Keyboard, Screen } from 'smartway-react-native-ui';
+import type { KeyboardActions, KeyboardState } from 'src/components/keyboard/Keyboard';
+
+interface Inputs {
+    upperInput: string;
+    bottomInput: string;
+}
+
+export const KeyboardPage = () => {
+    const [focusedInput, setFocusedInput] = useState<string>('');
+    const [value, setValue] = useState<KeyboardState>({ value: '', action: 'none' });
+    const [inputValues, setValues] = useState<Inputs>({ upperInput: '', bottomInput: '' });
+
+    const onChangeTextHandler = ({ value, action }: KeyboardState) => {
```

Comment:
I don't like this design: it exposes too much internal of the keyboard, and this logic will need to be rewritten of every screen where a Keyboard is required (pretty much every of our screens).

I don't have a good alternative either so I need a bit of time to think about it; and it would be great if the others could give their inputs too; my react knowledge are not enough.

Ping @GoMino @sGeeK44 @valentinmagrez @AudBrou

Reply:
Have you considered passing the `TextInputRN` component as a prop value to the `Keyboard` component, and let the `Keyboard` component handle this internally ?

Reply:
No, but at least i should try it

Reply:
Are there other types of inputs or is `TextInput` the only "base" component ?

Reply:
I tried with TextInput for the demo app,so i think yes, the component will be used from TextInput base